### PR TITLE
Ensure that a supplied vstest.console path is escape sequenced

### PR DESCRIPTION
**Pull request by AbhitejJohn**
## Description
A supplied path to vstest.console currently may have spaces and could break the logic we have in place. Fixed this to ensure we escape-sequence this path. This is required for a change on the Test Explorer side.

### CODE REVIEW

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs [hunk 1]
```diff
@@ -220,15 +220,24 @@ private string[] BuildArguments(ConsoleParameters parameters)
...
+                args.Insert(0, GetEscapeSequencedPath(vstestConsolePath));
             }
 
             return args.ToArray();
         }
 
         private string GetConsoleRunner()
+            => isNetCoreRunner ? ( string.IsNullOrEmpty(this.dotnetExePath) ? new DotnetHostHelper().GetDotnetPath() : this.dotnetExePath) : GetEscapeSequencedPath(this.vstestConsolePath);
+
+        private string GetEscapeSequencedPath(string path)
```
Comment (nohwnd) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs:232]: I never heard any path to be referred to as "escape sequenced". I would call this quoted path.

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs [hunk 2]
```diff
@@ -220,15 +220,24 @@ private string[] BuildArguments(ConsoleParameters parameters)
...
         {
-            return isNetCoreRunner ? ( string.IsNullOrEmpty(this.dotnetExePath) ? new DotnetHostHelper().GetDotnetPath() : this.dotnetExePath) : vstestConsolePath;
+            if (path == null || (path.StartsWith("\"") && path.EndsWith("\"")))
```
Comment (nohwnd) [inline src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs:234]:
```
if (path == null)
{
    return path;
}

return $"\"{path.Trim('"')}\"";
```
This seems to have less edge cases.

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs (hunk 1 above)
Comment (AbhitejJohn) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs:232]: oh looks like we have a few references to those words in this repo already: https://github.com/microsoft/vstest/search?q=Escape+sequence

**Comment by AbhitejJohn** [comment]
@nohwnd : Just a friendly reminder on this. Sounds like its good to go. I've addressed all your comments.

**Comment by AbhitejJohn** [comment]
@microsoft/vstest : hoping to get a review to merge this in.

**Review by Sanan07 — APPROVED** [review]
LGTM

File: src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs (hunk 1 above)
Comment (nohwnd) [reply src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs:232]: They refer to "escape sequence" which is well established, and means prepending special characters with escape character. Your method just adds quotes. Quotes would sometimes be escaped, but are themselves not escape sequence. 

I am not trying to make this unnecessarily difficult. But if Sanan thinks this is good, let's merge it then.

**Pull request merged**